netbox/client.go

```diff
@@ -38,16 +38,18 @@ type Client struct {
 	// Tenancy provides access to methods in NetBox's Tenancy API.
 	Tenancy *TenancyService
 
+	token  *string
 	u      *url.URL
 	client *http.Client
 }
 
 // NewClient returns a new instance of a NetBox client.  addr specifies the address
-// of the NetBox server, and client specifies an optional HTTP client to use
-// for requests.
+// of the NetBox server, token optionally specifies an api key to use,
+// and client specifies an optional HTTP client to use for requests.
 //
+// If token is nil, no Authentication header will be included with requests.
 // If client is nil, a default HTTP client will be used.
-func NewClient(addr string, client *http.Client) (*Client, error) {
+func NewClient(addr string, token *string, client *http.Client) (*Client, error) {
```

Comment:
For basic types, the zero value is more common for "don't set this parameter", instead of using a nil pointer.

IMO `if token == ""` that should mean "no authentication", though as mentioned, isn't authentication required for all calls now?